### ISSUE TITLE
Why is there no google-chrome?

### ISSUE THREAD

**Issue by rbnlffl** [lukesampson/scoop-extras#400]
Is there a reason you can install Firefox via Scoop but not Chrome? If not, I think it'd be pretty cool to have all the major 3rd party browsers (Chrome, Firefox, Opera (maybe Safari)) or even just Chrome installable. This would be sweet for us web developers used to unix.

![screencap](https://cloud.githubusercontent.com/assets/6851646/25942429/1cef30e4-363d-11e7-8a6c-b5ec49caf2ec.png)

**Comment by lukesampson** [comment]
Answer: because no one has submitted a [manifest](https://github.com/lukesampson/scoop/wiki/App-Manifests) yet! Would appreciate your pull request if you have time.

**Comment by Guard13007** [comment]
Going with the philosophy of `scoop`. it seems the version of Google Chrome packaged with the PortableApps.com platform should be used, but I would argue such a manifest should just install Chrome the regular way. Thoughts?

**Comment by lukesampson** [comment]
It looks like the PortableApps.com download is just a stub that downloads the program files. So, besides versioning info, I'm not sure what benefit that provides over downloading ChromeSetup.exe direct from Google.

The "standalone" version from Google doesn't appear to be easily extractable (e.g. `https://dl.google.com/tag/s/appguid%3D%7B8A69D345-D564-463C-AFF1-A69D9E530F96%7D%26iid%3D%7B8F53A4DD-FDE7-E6D6-C2D4-C2E1E9735C56%7D%26lang%3Den%26browser%3D4%26usagestats%3D0%26appname%3DGoogle%2520Chrome%26needsadmin%3Dprefers%26ap%3Dx64-stable/update2/installers/ChromeStandaloneSetup64.exe`)

Reading [here](https://reverseengineering.stackexchange.com/questions/4618/extracting-files-from-google-chrome-offline-installer) it sounds like Google have put measures in place to prevent the kind of minimalist install that Scoop is good at (download, extract, link).

This may be why no one has submitted Chrome to Scoop so far. Google seems intent on controlling the install and doing their own updates, and it looks difficult to circumvent and simplify this. A Scoop install that could overcome these difficulties wouldn't seem to provide users much benefit over the official installer.

If it even works, there might be issues with updates. Scoop would think the original version was installed while it has been updated in place by Google Update.

TLDR; abandon all hope. It'll be hard to get a working Scoop install, hard to maintain, and likely provide users a sub-par update experience. Would still accept a PR if you do want to proceed.

**Comment by rbnlffl** [comment]
Awh, bummer!

But then, in regards to Opera: It's a self-updating browser too, right? So if I initially install it via Scoop, but it then updates itself, how will Scoop handle that? Same thing for Firefox.

**Comment by lukesampson** [comment]
Scoop doesn't handle self-updating programs—they will be reported as out of date if they've been self-updated. It is a minus. But the difficult install with possible admin permissions, GUI install, and unknown side effects are the deal breaker.

**Comment by Guard13007** [comment]
@lukesampson The PortableApps packaging keeps everything self-contained by keeping track of and undoing changes to the host operating system between launches. It seems like it would be the only way to get things to work given the issue with Google explicitly preventing doing what scoop is trying to do.

I don't know if it's worth doing though.

**Comment by lukesampson** [comment]
I'm not sure if users would appreciate that PortableApps behavior—seems like unnecessary interference to me. I tend to agree it's not worth it.

**Comment by rbnlffl** [comment]
In that case I think not adding google-chrome sounds reasonable.

**Comment by janpio** [comment]
Is still situation still unchanged today? 

Chocolatey seems to have found a way to install Chrome, https://chocolatey.org/packages/GoogleChrome (check the scripts), is this because it only runs with admin rights?

**Comment by Ash258** [comment]
But chocolatey does not care, where you install application. Since chrome installer cannot be forced to change destination (for my tests few months ago, TARGETDIR, other generic variables are not working), it's not suitable to scoop behaviour. Also not everyone want to install using enterprise level installers with some default configs.

Alternatively you can use untrusted chrome offline installer made by users (like I do https://github.com/Ash258/scoop-Ash258/blob/master/bucket/Chrome.json), but it's not recommended for most of users.

**Comment by asd-collab** [comment]
https://github.com/lukesampson/scoop-extras/issues/400#issuecomment-301208707

changes are not persisted between upgrades. if you have old browser version installed and do `scoop update`, both for chrome and firefox. and the browsers don't allow importing from their previous versions, even though previous versions provide password exports. similarly for browser history and bookmarks.

and there are other issues. so avoid using scoop to install browsers.

---

asd-collab
